src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Http/FrameOfT.cs

```diff
@@ -40,26 +47,47 @@ public Frame(IHttpApplication<TContext> application, FrameContext frameContext)
 
                     while (!_requestProcessingStopping)
                     {
-                        var result = await Input.ReadAsync();
-                        var examined = result.Buffer.End;
-                        var consumed = result.Buffer.End;
+                        if (needBuffer)
+                        {
+                            result = await Input.ReadAsync();
+                            buffer = result.Buffer;
+                            needBuffer = false;
+                        }
+
+                        var needAdvance = false;
 
                         try
                         {
-                            ParseRequest(result.Buffer, out consumed, out examined);
+                            ParseRequest(ref buffer, out consumed, out examined);
+
+                            if (buffer.Length == 0)
```

Comment:
Hmmmm, after thinking about this for a bit. It could happen. We could check if `examined == buffer.End` and set needBuffer to true so it doesn't spin. Will add a unit test for this.